distribution/src/bin/elasticsearch

```diff
@@ -17,15 +17,30 @@ source "`dirname "$0"`"/elasticsearch-env
 
 CHECK_KEYSTORE=true
 DAEMONIZE=false
-for option in "$@"; do
-  case "$option" in
-    -h|--help|-V|--version)
-      CHECK_KEYSTORE=false
-      ;;
-    -d|--daemonize)
-      DAEMONIZE=true
-      ;;
-  esac
+ENROLL_TO_CLUSTER=false
+# Store original arg array as we will be shifting through it below
+ARG_LIST=($@)
+
+while [ $# -gt 0 ]; do
+   if [[ $1 == "--enrollment-token" ]]; then
+     ENROLL_TO_CLUSTER=true
+     ENROLLMENT_TOKEN="$2"
```

Comment:
Can we validate that this has been supplied?

Reply:
If not, $2 is empty and we detect this and handle/fail accordingly in EnrollNodeToCluster. I'm testing for this in https://github.com/elastic/elasticsearch/pull/77718/files#diff-e15db1cf097589a885ce51af8af3cdc8c2a3bf009b89cbc8065b2816a66f1bd2R194. 

I have nothing against doing it here too but it seems we're not doing much validation of parameters in the bash script anyhow, plus doing it in Java allows us to reuse the validation in the windows batch script too